lib/rpi.pm

```diff
@@ -0,0 +1,37 @@
+# SUSE's openQA tests
+#
+# Copyright SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+package rpi;
+
+use warnings;
+use strict;
+
+use Exporter 'import';
+use testapi;
+use power_action_utils 'power_action';
+
+our @EXPORT = 'enable_tpm_slb9670';
+
+# Enable TPM on Raspberry Pi
+# https://en.opensuse.org/HCL:Raspberry_Pi3_TPM
+sub enable_tpm_slb9670 {
+    my $module = "tpm_tis_spi";
+
+    assert_script_run('echo -e "dtparam=spi=on\ndtoverlay=tpm-slb9670" >> /boot/efi/extraconfig.txt');
+    assert_script_run('cat /boot/efi/extraconfig.txt');
+    assert_script_run("echo '$module' > /etc/modules-load.d/tpm.conf");
+    power_action('reboot', textmode => 1);
+
+    # Add some timeout to wait for reboot
+    sleep(60);
```

Comment:
i know that this is moved here and it is not coming from you but i am wondering if it would be better to replace it with `wait_serial`? wdyt?

Reply:
I can try that.

Reply:
not wait serial but something like `wait_boot` should be here

Reply:
Well, since `wait_boot` is skipped for JeOS

https://github.com/pevik/os-autoinst-distri-opensuse/blob/5cbccd4f49840ef36cf07bff474c03c83ad90b61/tests/kernel/install_ltp.pm#L358-L360

and none of these helped:
```
$self->wait_boot(nologin => 1, bootloader_time => 60);
$self->wait_boot;
```
Because `wait_boot()` uses `assert_screen()`

https://github.com/pevik/os-autoinst-distri-opensuse/blob/5cbccd4f49840ef36cf07bff474c03c83ad90b61/lib/opensusebasetest.pm#L1175-L1187. I suppose we need some support for RPI in openQA, but that's out of my scope.

Reply:
I think the `sleep` is the only option here since it is a headless system, reached via ssh only.

Reply:
Given that I suggest to merge. @b10n1k @mimi1vx @tjyrinki ?

Reply:
let s move on